cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -478,6 +478,38 @@ std::shared_ptr<PointCloud> PointCloud::RandomDownSample(
     return SelectByIndex(indices);
 }
 
+std::shared_ptr<PointCloud> PointCloud::GetFarthestPointSample(
+        size_t num_samples) const {
+    if (num_samples <= 0) {
+        utility::LogError(
+                "[GetFarthestPointSample] Illegal number of points, the number "
+                "should larger than 0");
+    }
+    if (num_samples >= points_.size()) {
+        return std::make_shared<PointCloud>(*this);
+    } else {
+        std::vector<size_t> indices;
+        indices.resize(num_samples);
+
+        const size_t num = points_.size();
+        std::vector<double> distance(num, 1e10);
+        size_t farthest_index = 0;
+        for (size_t i = 0; i < num_samples; i++) {
+            indices[i] = farthest_index;
+            auto &selected = points_[farthest_index];
+            for (size_t j = 0; j < num; j++) {
+                double dist = (points_[j] - selected).array().square().sum();
+                distance[j] = std::min(distance[j], dist);
+            }
+            farthest_index = std::distance(
```

Comment:
Try merging this inside the `for (size_t j = 0; j < num; j++) {` for loop. You may benchmark it in the python example and check the run time. 

E.g., 
```diff
+            double max_dist = 0;
             for (size_t j = 0; j < num; j++) {
                 double dist = (points_[j] - selected).array().square().sum();
                 distance[j] = std::min(distance[j], dist);
+                if (distance[j] > max_dist) {
+                    max_dist = distance[j];
+                    farthest_index = j;
+                }
             }
```

Reply:
Thanks for your inspiration! This modification makes it about 30% faster than original one